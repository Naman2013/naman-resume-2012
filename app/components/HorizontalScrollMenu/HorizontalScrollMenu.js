import React from 'react';
import PropTypes from 'prop-types';
import style from './HorizontalScrollMenu.style';

const HorizontalScrollMenu = () => (
  <div>
    <ul>
      <li>
        <a href="#">Overview</a>
      </li>
      <li>
        <a href="#">Quests</a>
      </li>
      <li>
        <a href="#">Ask anyone</a>
      </li>
      <li>
        <a href="#">Chicken salad BLT</a>
      </li>
    </ul>
    <style jsx>{style}</style>
  </div>
);

export default HorizontalScrollMenu;
