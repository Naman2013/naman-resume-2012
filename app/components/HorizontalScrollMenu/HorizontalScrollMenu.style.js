import css from 'styled-jsx/css';
import { chillBlue } from 'styles/variables/colors_tiles_v4';
import { primaryFont } from 'styles/variables/fonts';
import { resetMarginPadding } from 'styles/variables/utils';

export default css`
  ul {
    ${resetMarginPadding}
    padding: 0 35px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 68px;
    list-style-type: none;
    background: linear-gradient(to bottom, #0b121b 0%,#263344 74%);
    overflow: hidden;
  }

  a {
    ${resetMarginPadding}
    display: inline-block;
    text-align: center;
    text-decoration: none;
    text-transform: uppercase;
    color: ${chillBlue};
    font-family: ${primaryFont};
    font-size: 11px;
    font-weight: 800;
    letter-spacing: 1.4px;
    min-width: 150px;
  }
`;
