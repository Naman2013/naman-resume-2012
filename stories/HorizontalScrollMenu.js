import React from 'react';
import { storiesOf } from '@storybook/react';
import HorizontalScrollMenu from '../app/components/HorizontalScrollMenu';

storiesOf('HorizontalScrollMenu', module)
  .add('Default', () => (<HorizontalScrollMenu />));
