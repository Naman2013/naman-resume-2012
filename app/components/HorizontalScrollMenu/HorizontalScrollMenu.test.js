import React from 'react';
import { shallow } from 'enzyme';
import HorizontalScrollMenu from './HorizontalScrollMenu';

describe('HorizontalScrollMenu', () => {
  const wrapper = shallow(<HorizontalScrollMenu />);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
