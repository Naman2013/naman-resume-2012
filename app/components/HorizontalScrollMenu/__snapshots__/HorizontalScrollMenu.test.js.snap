// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalScrollMenu should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HorizontalScrollMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ul
          className="jsx-3286086166"
        >
          <li
            className="jsx-3286086166"
          >
            <a
              className="jsx-3286086166"
              href="#"
            >
              Overview
            </a>
            <span
              className="jsx-3286086166 triangle"
            >
              <Triangle
                height={8}
                theme={
                  Object {
                    "color": "white",
                  }
                }
                width={10}
              />
            </span>
          </li>
          <li
            className="jsx-3286086166"
          >
            <a
              className="jsx-3286086166"
              href="#"
            >
              Quests
            </a>
          </li>
          <li
            className="jsx-3286086166"
          >
            <a
              className="jsx-3286086166"
              href="#"
            >
              Ask anyone
            </a>
          </li>
          <li
            className="jsx-3286086166"
          >
            <a
              className="jsx-3286086166"
              href="#"
            >
              Chicken salad BLT
            </a>
          </li>
        </ul>,
        <JSXStyle
          css={
            Array [
              "ul.jsx-3286086166{
  margin: 0;
  padding: 0;
 padding:0 30px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;list-style-type:none;background:linear-gradient(to bottom,#0b121b 0%,#263344 74%);overflow-x:scroll;}",
              "ul.jsx-3286086166::-webkit-scrollbar{width:0px;}",
              "ul.jsx-3286086166::-webkit-scrollbar-track-piece{background-color:transparent;border-radius:6px;display:none;}",
              "li.jsx-3286086166{position:relative;}",
              "a.jsx-3286086166{
  margin: 0;
  padding: 0;
 padding:30px 0;padding-bottom:20px;min-width:120px;display:inline-block;text-align:center;-webkit-text-decoration:none;text-decoration:none;text-transform:uppercase;color:#9cb4d2;font-family:brandon-grotesque, sans-serif;font-size:11px;font-weight:800;-webkit-letter-spacing:1.4px;-moz-letter-spacing:1.4px;-ms-letter-spacing:1.4px;letter-spacing:1.4px;white-space:nowrap;}",
              ".triangle.jsx-3286086166{display:block;position:absolute;top:52px;left:50%;-webkit-transform:translate(-50%);-ms-transform:translate(-50%);transform:translate(-50%);}",
            ]
          }
          styleId="3286086166"
        />,
      ],
      "className": "jsx-3286086166",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="jsx-3286086166"
            >
              <a
                className="jsx-3286086166"
                href="#"
              >
                Overview
              </a>
              <span
                className="jsx-3286086166 triangle"
              >
                <Triangle
                  height={8}
                  theme={
                    Object {
                      "color": "white",
                    }
                  }
                  width={10}
                />
              </span>
            </li>,
            <li
              className="jsx-3286086166"
            >
              <a
                className="jsx-3286086166"
                href="#"
              >
                Quests
              </a>
            </li>,
            <li
              className="jsx-3286086166"
            >
              <a
                className="jsx-3286086166"
                href="#"
              >
                Ask anyone
              </a>
            </li>,
            <li
              className="jsx-3286086166"
            >
              <a
                className="jsx-3286086166"
                href="#"
              >
                Chicken salad BLT
              </a>
            </li>,
          ],
          "className": "jsx-3286086166",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="jsx-3286086166"
                  href="#"
                >
                  Overview
                </a>,
                <span
                  className="jsx-3286086166 triangle"
                >
                  <Triangle
                    height={8}
                    theme={
                      Object {
                        "color": "white",
                      }
                    }
                    width={10}
                  />
                </span>,
              ],
              "className": "jsx-3286086166",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Overview",
                  "className": "jsx-3286086166",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Overview",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Triangle
                    height={8}
                    theme={
                      Object {
                        "color": "white",
                      }
                    }
                    width={10}
                  />,
                  "className": "jsx-3286086166 triangle",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "height": 8,
                    "theme": Object {
                      "color": "white",
                    },
                    "width": 10,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="jsx-3286086166"
                href="#"
              >
                Quests
              </a>,
              "className": "jsx-3286086166",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Quests",
                "className": "jsx-3286086166",
                "href": "#",
              },
              "ref": null,
              "rendered": "Quests",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="jsx-3286086166"
                href="#"
              >
                Ask anyone
              </a>,
              "className": "jsx-3286086166",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ask anyone",
                "className": "jsx-3286086166",
                "href": "#",
              },
              "ref": null,
              "rendered": "Ask anyone",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="jsx-3286086166"
                href="#"
              >
                Chicken salad BLT
              </a>,
              "className": "jsx-3286086166",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Chicken salad BLT",
                "className": "jsx-3286086166",
                "href": "#",
              },
              "ref": null,
              "rendered": "Chicken salad BLT",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "ul.jsx-3286086166{
  margin: 0;
  padding: 0;
 padding:0 30px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;list-style-type:none;background:linear-gradient(to bottom,#0b121b 0%,#263344 74%);overflow-x:scroll;}",
            "ul.jsx-3286086166::-webkit-scrollbar{width:0px;}",
            "ul.jsx-3286086166::-webkit-scrollbar-track-piece{background-color:transparent;border-radius:6px;display:none;}",
            "li.jsx-3286086166{position:relative;}",
            "a.jsx-3286086166{
  margin: 0;
  padding: 0;
 padding:30px 0;padding-bottom:20px;min-width:120px;display:inline-block;text-align:center;-webkit-text-decoration:none;text-decoration:none;text-transform:uppercase;color:#9cb4d2;font-family:brandon-grotesque, sans-serif;font-size:11px;font-weight:800;-webkit-letter-spacing:1.4px;-moz-letter-spacing:1.4px;-ms-letter-spacing:1.4px;letter-spacing:1.4px;white-space:nowrap;}",
            ".triangle.jsx-3286086166{display:block;position:absolute;top:52px;left:50%;-webkit-transform:translate(-50%);-ms-transform:translate(-50%);transform:translate(-50%);}",
          ],
          "styleId": "3286086166",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            className="jsx-3286086166"
          >
            <li
              className="jsx-3286086166"
            >
              <a
                className="jsx-3286086166"
                href="#"
              >
                Overview
              </a>
              <span
                className="jsx-3286086166 triangle"
              >
                <Triangle
                  height={8}
                  theme={
                    Object {
                      "color": "white",
                    }
                  }
                  width={10}
                />
              </span>
            </li>
            <li
              className="jsx-3286086166"
            >
              <a
                className="jsx-3286086166"
                href="#"
              >
                Quests
              </a>
            </li>
            <li
              className="jsx-3286086166"
            >
              <a
                className="jsx-3286086166"
                href="#"
              >
                Ask anyone
              </a>
            </li>
            <li
              className="jsx-3286086166"
            >
              <a
                className="jsx-3286086166"
                href="#"
              >
                Chicken salad BLT
              </a>
            </li>
          </ul>,
          <JSXStyle
            css={
              Array [
                "ul.jsx-3286086166{
  margin: 0;
  padding: 0;
 padding:0 30px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;list-style-type:none;background:linear-gradient(to bottom,#0b121b 0%,#263344 74%);overflow-x:scroll;}",
                "ul.jsx-3286086166::-webkit-scrollbar{width:0px;}",
                "ul.jsx-3286086166::-webkit-scrollbar-track-piece{background-color:transparent;border-radius:6px;display:none;}",
                "li.jsx-3286086166{position:relative;}",
                "a.jsx-3286086166{
  margin: 0;
  padding: 0;
 padding:30px 0;padding-bottom:20px;min-width:120px;display:inline-block;text-align:center;-webkit-text-decoration:none;text-decoration:none;text-transform:uppercase;color:#9cb4d2;font-family:brandon-grotesque, sans-serif;font-size:11px;font-weight:800;-webkit-letter-spacing:1.4px;-moz-letter-spacing:1.4px;-ms-letter-spacing:1.4px;letter-spacing:1.4px;white-space:nowrap;}",
                ".triangle.jsx-3286086166{display:block;position:absolute;top:52px;left:50%;-webkit-transform:translate(-50%);-ms-transform:translate(-50%);transform:translate(-50%);}",
              ]
            }
            styleId="3286086166"
          />,
        ],
        "className": "jsx-3286086166",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="jsx-3286086166"
              >
                <a
                  className="jsx-3286086166"
                  href="#"
                >
                  Overview
                </a>
                <span
                  className="jsx-3286086166 triangle"
                >
                  <Triangle
                    height={8}
                    theme={
                      Object {
                        "color": "white",
                      }
                    }
                    width={10}
                  />
                </span>
              </li>,
              <li
                className="jsx-3286086166"
              >
                <a
                  className="jsx-3286086166"
                  href="#"
                >
                  Quests
                </a>
              </li>,
              <li
                className="jsx-3286086166"
              >
                <a
                  className="jsx-3286086166"
                  href="#"
                >
                  Ask anyone
                </a>
              </li>,
              <li
                className="jsx-3286086166"
              >
                <a
                  className="jsx-3286086166"
                  href="#"
                >
                  Chicken salad BLT
                </a>
              </li>,
            ],
            "className": "jsx-3286086166",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    className="jsx-3286086166"
                    href="#"
                  >
                    Overview
                  </a>,
                  <span
                    className="jsx-3286086166 triangle"
                  >
                    <Triangle
                      height={8}
                      theme={
                        Object {
                          "color": "white",
                        }
                      }
                      width={10}
                    />
                  </span>,
                ],
                "className": "jsx-3286086166",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Overview",
                    "className": "jsx-3286086166",
                    "href": "#",
                  },
                  "ref": null,
                  "rendered": "Overview",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Triangle
                      height={8}
                      theme={
                        Object {
                          "color": "white",
                        }
                      }
                      width={10}
                    />,
                    "className": "jsx-3286086166 triangle",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "height": 8,
                      "theme": Object {
                        "color": "white",
                      },
                      "width": 10,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="jsx-3286086166"
                  href="#"
                >
                  Quests
                </a>,
                "className": "jsx-3286086166",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Quests",
                  "className": "jsx-3286086166",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Quests",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="jsx-3286086166"
                  href="#"
                >
                  Ask anyone
                </a>,
                "className": "jsx-3286086166",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Ask anyone",
                  "className": "jsx-3286086166",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Ask anyone",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="jsx-3286086166"
                  href="#"
                >
                  Chicken salad BLT
                </a>,
                "className": "jsx-3286086166",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Chicken salad BLT",
                  "className": "jsx-3286086166",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Chicken salad BLT",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "ul.jsx-3286086166{
  margin: 0;
  padding: 0;
 padding:0 30px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;list-style-type:none;background:linear-gradient(to bottom,#0b121b 0%,#263344 74%);overflow-x:scroll;}",
              "ul.jsx-3286086166::-webkit-scrollbar{width:0px;}",
              "ul.jsx-3286086166::-webkit-scrollbar-track-piece{background-color:transparent;border-radius:6px;display:none;}",
              "li.jsx-3286086166{position:relative;}",
              "a.jsx-3286086166{
  margin: 0;
  padding: 0;
 padding:30px 0;padding-bottom:20px;min-width:120px;display:inline-block;text-align:center;-webkit-text-decoration:none;text-decoration:none;text-transform:uppercase;color:#9cb4d2;font-family:brandon-grotesque, sans-serif;font-size:11px;font-weight:800;-webkit-letter-spacing:1.4px;-moz-letter-spacing:1.4px;-ms-letter-spacing:1.4px;letter-spacing:1.4px;white-space:nowrap;}",
              ".triangle.jsx-3286086166{display:block;position:absolute;top:52px;left:50%;-webkit-transform:translate(-50%);-ms-transform:translate(-50%);transform:translate(-50%);}",
            ],
            "styleId": "3286086166",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
